src/httpClient.ts

```diff
@@ -270,3 +276,35 @@ function make_request_timing(
       deserialize_end && body_read_end ? deserialize_end - body_read_end : 0,
   };
 }
+
+function convertHeadersType(
+  headers: Record<string, string | string[] | undefined>,
+): Record<string, string> {
+  for (const key in headers) {
+    const v = headers[key];
+    if (v === undefined) {
+      delete headers[key];
+    } else if (Array.isArray(v)) {
+      headers[key] = v[0];
+    }
+  }
+  return headers as Record<string, string>;
+}
+
+async function consumeResponseText(response: Dispatcher.ResponseData): Promise<{
+  body_text: string;
+  body_read_end: number;
+}> {
+  if (response.headers["content-encoding"] == "gzip") {
+    let body_buffer = await response.body.arrayBuffer();
+    let body_read_end = performance.now();
+
+    let gunzip_buffer = await gunzip_async(body_buffer);
+    let body_text = gunzip_buffer.toString(); // is there a better way?
```

Comment:
undici-fetch uses [TextDecoder](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder) but idk if it's any better

Reply:
Doubt it... 

As per your comment, you can `JSON.parse` buffer directly, but doesn't seem like that makes any difference, [see code](https://gist.github.com/sirupsen/603855eef0fbe9c576beddabc4266a60)